apps/console/src/features/identity-providers/components/settings/general-settings.tsx

```diff
@@ -362,9 +364,12 @@ export const GeneralSettings: FunctionComponent<GeneralSettingsInterface> = (
                                 primaryAction={ t("common:confirm") }
                                 secondaryAction={ t("common:cancel") }
                                 onSecondaryActionClick={ (): void => setShowDeleteConfirmationModal(false) }
-                                onPrimaryActionClick={
-                                    (): void => handleIdentityProviderDelete()
-                                }
+                                onPrimaryActionClick={ (): void => {
+                                    setLoading(true);
+                                    handleIdentityProviderDelete().finally(() => {
```

Comment:
Abstracting the function can be done as same as the above comment

Reply:
Addressed via af36e121eed12e6c0f06003fee7e01072b7c4ee6